internal/api/impl/auth/oauth.go

```diff
@@ -268,6 +297,40 @@ func (e *oAuthEndpoint) requestUserInfos(ctx echo.Context, token *oauth2.Token)
 	return nil, nil
 }
 
+// buildUserInfo will collect all the information needed to create/update the user in database.
+// Two source of information are considered:
+//   - token: the access_token generated from the oauth 2.0 provider.
+//   - api: the body of a possible request to a user infos url
+//     if a /userinfo endpoint has been configured, some information can be collected from there
+//
+// In the future, this method could evolve taking some configuration parameters to allow more customization
+// on claim paths/JSON paths used.
+func (e *oAuthEndpoint) buildUserInfo(token *oauth2.Token, body []byte) (externalUserInfo, error) {
+	userInfos := oauthUserInfo{}
+	// Parse token
+	var errToken error
+	if token != nil {
+		// TODO(cegarcia): Is it really necessary?, claims that we are searching for are id_token claims,
+		//  and if there is an id token, then it is OIDC
```

Comment:
Putting back into draft mode in order to resolve this comment. 
I will try more oauth providers as for now only github has been tried and doesn't have jwt access_token neither id_token. 
I have a good hope that this token parsing is actually not necessary at all.

Reply:
I made the choice to not parse the token here and rely only on API /userinfos data of the provider to retrieve needful information. This will be acceptable as first shot and we'll see with usages if we need to improve.

Anyway, with generic oauth, access_token is not guaranteed to be a jwt token neither to contain the right claims at the right place. A true id_token would guarantee that but if we receive id_token, then it is OIDC.

From what I see in the web, the mainstream will be more and more OIDC and this is what we should push for as much as possible.